src/init/features/apphosting/index.ts

```diff
@@ -136,6 +157,69 @@ export async function createBackend(
     pollerName: `create-${projectId}-${location}-${backendId}`,
     operationResourceName: op.name,
   });
-
   return backend;
 }
+
+/**
+ * Only lowercase, digits, and hyphens; must begin with letter, and cannot end with hyphen
+ */
+function generateBuildId(n = 6) {
+  const letters = "abcdefghijklmnopqrstuvwxyz";
+  const allChars = "01234567890-abcdefghijklmnopqrstuvwxyz";
+  let id = letters[Math.floor(Math.random() * letters.length)];
+  for (let i = 1; i < n; i++) {
+    const idx = Math.floor(Math.random() * allChars.length);
+    id += allChars[idx];
+  }
+  return id;
+}
+
+/**
+ * Create (and waits for) a new build.
+ */
+export async function createBuild(
+  projectId: string,
+  location: string,
+  backendId: string,
+  buildInput: Omit<BuildInput, "name">
+): Promise<Build> {
+  logBullet("Creating a new build... this may take a few minutes.");
+  const buildId = generateBuildId();
```

Comment:
is build id a required field? Can we let the server create it for us?

Reply:
Unfortunately, yes for now.